### PR TITLE
Enable prototype methods for JavaScript objects in UDFs

### PR DESCRIPTION
**Pull request by proddata**
## Summary of the changes / Why this improves CrateDB
Enable prototype methods for JavaScript objects in UDFs

---

By default GraalVM does map Java arrays to shallow JavaScript arrays with an empty prototype objects [1]. This means, that if you want to call e.g. array methods like 

```js
myArray.sort((a,b) => b - a)
```

one needs to explicitly call them like

```js
Array.prototype.call.sort(myArray, (a,b) => b - a)
```

This behaviour can be changed by enabling the `js.foreign-object-prototype` option, which has been moved from the experimental stage in version 20.3.0. 

> When enabled, objects on the JavaScript side get the most prototype (e.g. Array.prototype, Function.prototype, Object.prototype) set and can thus behave more similarly to native JavaScript objects of the respective type. Normal JavaScript precedence rules apply here, e.g., own properties (of the Java object in that case) take precedence over and hide properties from the prototype.

This change should make the writing of JavaScript UDFs more straight forward for new users.

[1] https://www.graalvm.org/reference-manual/js/FAQ/#builtin-functions-like-arraymap-or-fnapply-are-not-available-on-non-javascript-objects-like-proxyarrays-from-java

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: docs/general/user-defined-functions.rst [hunk 1]
```diff
@@ -392,45 +392,8 @@ is returned::
...
-
-.. code-block:: js
-
-    function array_join(a, b) {
-        return Array.prototype.join.call(a, b);
-    }
-
-You must do it like this because arguments are not passed as ``Array`` objects,
-and so do not have the associated prototype methods available. Arguments are instead
-passed as array-like objects.
```
[the remark this answers is not in the document]
Comment (proddata) [reply docs/general/user-defined-functions.rst:424]: I hope this should be resolved with the new **note**

File: docs/general/user-defined-functions.rst [hunk 2]
```diff
@@ -392,45 +392,8 @@ is returned::
...
-use, such as `join`_, `map`_, `sort`_, `slice`_, `reduce`_, and so on.
-
-Normally, you can call these methods directly from an ``Array`` object, like
-so:
-
-.. code-block:: js
-
-    function array_join(a, b) {
-        return a.join(b);
-    }
```
[the remark this answers is not in the document]
Comment (proddata) [reply docs/general/user-defined-functions.rst:411]: added a not in the **Javascript** section

**Review by mfussenegger — APPROVED** [review]

**Comment by mfussenegger** [comment]
ok to test

**Comment by mfussenegger** [comment]
retest this please

**Pull request merged**